cl_hubeau/water_services/utils.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Convenience functions for water services indicators.
+"""
+
+import geopandas as gpd
+import pandas as pd
+from tqdm import tqdm
+
+from cl_hubeau.water_services.water_services_scraper import WaterServicesSession
+from cl_hubeau import _config
+from cl_hubeau.utils import get_departements_from_regions, get_regions, prepare_kwargs_loops
+from datetime import datetime
+
+def get_all_communes(code_region=None, **kwargs) -> gpd.GeoDataFrame:
+    """
+    Gets the services for every french city.
+
+    Parameters
+    ----------
+    **kwargs :
+        kwargs passed to WaterServicesSession.get_communes (hence mostly intended
+        for hub'eau API's arguments). Do not use `format` or `code_commune`
+        as they are set by the current function.
+
+    Returns
+    -------
+    results : gpd.GeoDataFrame
+        GeoDataFrame of water services indicators for every french city.
+
+    """
+    if(not kwargs.get("code_commune")):
+        if code_region :
+            deps = get_departements_from_regions(code_region)
+        elif kwargs.get("code_departement") :
+            deps = kwargs.get("code_departement")
+        else :
+            deps = get_departements_from_regions(get_regions(True))
+    with WaterServicesSession() as session :
+        if(kwargs.get("code_commune")) :
+            results = [session.get_communes(
+                code_commune=kwargs.get("code_commune")
+                )]
+        else:
+            results = [
+                session.get_communes(
+                    code_departement=get_departements_from_regions(dep), detail_service=True, format="geojson", **kwargs
+                )
+                for dep in tqdm(
+                    deps,
+                    desc="querying regions for communes",
+                    leave=_config["TQDM_LEAVE"],
+                    position=tqdm._get_free_pos(),
+                )
+            ]
+    results = [x[0].dropna(axis=1, how="all") for x in results if not x[0].empty]
+    results = gpd.pd.concat(results, ignore_index=True)
```

Comment:
I don't think that would work well with `detail_service=True`. In that case, we should need to concat both dataframes' lists individualy.